### PR TITLE
Issue #3286: Replace deprecated asList() in Web App Generator Test

### LINKED ISSUE #3286: WebAppGeneratorTest :  Replace AssertJ's deprecated `asList()` DSL method with `asInstanceOf(InstanceOfAssertFactories.list(type.class))`
### Component

JKube Kit

### Task description

In the latest versions of Assertj `asList()` DSL method has been marked as deprecated:

https://github.com/eclipse-jkube/jkube/blob/e1cde0bd0d4ab1a659db7b1f7a884705048b27de/jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/WebAppGeneratorTest.java#L129

users are advised to use `asInstanceOf(InstanceOfAssertFactories.list(type.class))` instead, we need to update `asList()` calls with this. 

### Expected Behavior
AssertJ's `asList()` is replaced with `asInstanceOf(InstanceOfAssertFactories.list(type.class))`

### Acceptance Criteria
- [ ] `asList()` DSL method call is replaced with  `asInstanceOf(InstanceOfAssertFactories.list(type.class))` in all places

### PR DESCRIPTION
**Pull request by ShivamChavan01**
## Description

Fixes #3286
Replace deprecated asList() in Web App Generator Test

## Type of change

 - [x] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.dev/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [x] My code follows the style guidelines of this project
 - [x] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [x] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [x] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/WebAppGeneratorTest.java [hunk 1]
```diff
@@ -23,10 +23,10 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Properties;
+
+import org.assertj.core.api.InstanceOfAssertFactories;
 import org.eclipse.jkube.generator.api.GeneratorContext;
-import org.eclipse.jkube.kit.common.JavaProject;
-import org.eclipse.jkube.kit.common.KitLogger;
-import org.eclipse.jkube.kit.common.Plugin;
+import org.eclipse.jkube.kit.common.*;
```
Comment (rohanKanojia) [inline jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/WebAppGeneratorTest.java:29]: could you please remove these wildcard imports?

Comment (rohanKanojia) [reply jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/WebAppGeneratorTest.java]: Sorry for not being clear, I was referring to getting rid of wildcard style imports (`.*`) and keeping single imports, now build is failing due to missing imports

```
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.13.0:testCompile (default-testCompile) on project jkube-kit-generator-webapp: Compilation failure: Compilation failure: 
[ERROR] /home/jenkins/agent/workspace/Build_Dependencies_PR-3302/jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/WebAppGeneratorTest.java:[131,83] cannot find symbol
[ERROR]   symbol:   class AssemblyFile
[ERROR]   location: class org.eclipse.jkube.generator.webapp.WebAppGeneratorTest.CustomizeAssembly
[ERROR] /home/jenkins/agent/workspace/Build_Dependencies_PR-3302/jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/WebAppGeneratorTest.java:[177,83] cannot find symbol
[ERROR]   symbol:   class AssemblyFile
[ERROR]   location: class org.eclipse.jkube.generator.webapp.WebAppGeneratorTest.CustomizeAssembly
[ERROR] -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.13.0:testCompile (default-testCompile) on project jkube-kit-generator-webapp: Compilation failure
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:375)
```

Comment (ShivamChavan01) [reply jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/WebAppGeneratorTest.java]: Sorry @rohanKanojia   I didn't Cross check with the imports previously so i made changes  and   now Cross Checked with all the imports i dont see any error now

Comment (rohanKanojia) [reply jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/WebAppGeneratorTest.java]: @ShivamChavan01 : okay, it's better to verify that the build is passing locally before pushing changes.

Comment (ShivamChavan01) [reply jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/WebAppGeneratorTest.java]: image

 All Done

**Review by rohanKanojia — APPROVED** [review]

**Review by manusa — APPROVED** [review]
LGTM, thx!

**Pull request merged**